### PR TITLE
Fixed Bonding tests for infiniband based interface

### PR DESCRIPTION
**Pull request by manvanthar**
Fixed Bonding tests for infiniband based interfaces

Signed-off-by: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>

### CODE REVIEW

File: io/net/bonding.py [hunk 1]
```diff
@@ -96,6 +96,10 @@ def setUp(self):
                                                 default=False)
         self.peer_wait_time = self.params.get("peer_wait_time", default=5)
         self.sleep_time = int(self.params.get("sleep_time", default=5))
+        self.ib = 'N'
+        if self.host_interface[0:2] == 'ib':
```
Comment (narasimhan-v) [inline io/net/bonding.py]: Assumption is infiniband interfaces start only with ib ?
Are there any other cases ?

File: io/net/bonding.py [hunk 2]
```diff
@@ -96,6 +96,10 @@ def setUp(self):
                                                 default=False)
         self.peer_wait_time = self.params.get("peer_wait_time", default=5)
         self.sleep_time = int(self.params.get("sleep_time", default=5))
+        self.ib = 'N'
```
Comment (narasimhan-v) [inline io/net/bonding.py]: This can be False / True, so that the checking later can be
`if self.ib` instead of `if self.ib == 'Y'`

**Comment by narasimhan-v** [comment]
@vaishnavibhat review please

File: io/net/bonding.py (hunk 1 above)
Comment (manvanthar) [reply io/net/bonding.py]: Yeah Atleast hasn't changed in last 10 yrs.

File: io/net/bonding.py (hunk 2 above)
Comment (manvanthar) [reply io/net/bonding.py]: ok can be done.

File: io/net/bonding.py [hunk 3]
```diff
@@ -172,6 +176,17 @@ def setup_ip(self):
...
+    def bond_ib_conf(self, arg1, arg2):
+        '''
+        configure slaves for IB cards
+        '''
+        if arg2 == "ATTACH":
+            cmd = 'ifenslave %s %s;' % (self.bond_name, arg1)
+        else:
+            cmd = 'ifenslave %s -d %s;' % (self.bond_name, arg1)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.log.info("unable to %s IB interface %s", (arg2, arg1))
```
Comment (abdhaleegit) [inline io/net/bonding.py]: @manvanthar  should the test WARN or FAIL if unable to attach ?

**Comment by manvanthar** [comment]
Fixed in PR #1378

**Pull request closed without merging**